src/collectors/mod.rs

```diff
@@ -3,10 +3,11 @@ pub use crate::collectors::github_rate_limit::GitHubRateLimit;
 
 use crate::MetricProvider;
 use futures::FutureExt;
+use std::borrow::Borrow;
 
 // register collectors for metrics gathering
 pub async fn register_collectors(p: &MetricProvider) -> Result<(), prometheus::Error> {
-    GitHubRateLimit::new()
+    GitHubRateLimit::new(p.config.borrow())
```

Comment:
Is there a reason why you called `.borrow()` instead of using `&p.config`?

Reply:
not really, I suppose my autocomplete is really smart, TAB, TAB 😅